src/app/modules/form/admin-form/admin-form.service.ts

```diff
@@ -502,3 +504,59 @@ export const updateForm = (
     return transformMongoError(error)
   })
 }
+
+export const updateFormSettings = (
+  originalForm: IPopulatedForm,
+  body: SettingsUpdateBody,
+): ResultAsync<
+  Partial<Pick<IFormDocument, keyof SettingsUpdateBody>>,
+  | DatabaseError
+  | DatabaseValidationError
+  | DatabaseConflictError
+  | DatabasePayloadSizeError
+> => {
+  const updateKeys = Object.keys(body) as (keyof SettingsUpdateBody)[]
+  const newForm = updateKeys.reduce((accumulatedForm, key) => {
+    switch (key) {
+      // Non-object patches, assign key to form.
+      case 'authType':
+      case 'emails':
+      case 'esrvcId':
+      case 'hasCaptcha':
+      case 'inactiveMessage':
+      case 'status':
+      case 'submissionLimit':
+      case 'title': {
+        set(accumulatedForm, key, body[key])
+        break
+      }
+      // Object patches, deep mutate.
+      case 'webhook': {
+        merge(accumulatedForm[key], body[key])
+        break
+      }
+      default:
+        // Purely compile time check for missed keys. No need to throw or return
+        // so unknown keys will just be skipped if that somehow happens during
+        // runtime.
+        new UnreachableCaseError(key)
+    }
+    return accumulatedForm
+  }, originalForm)
+
+  return ResultAsync.fromPromise(newForm.save(), (error) => {
+    logger.error({
+      message: 'Error encountered while updating form',
+      meta: {
+        action: 'updateFormSettings',
+        originalForm,
+        // Body is not logged in case sensitive data such as emails are stored.
+        updateKeys,
+      },
+      error,
+    })
+
+    return transformMongoError(error)
+    // Only return subset of original form that were updated.
+  }).map((updatedForm) => pick(updatedForm, updateKeys))
```

Comment:
We should return the Settings object as a single resource, so that the client always has the most updated representation.

Reply:
I see, so whole object picked by SettingsKeys instead of just the changed settings

Reply:
resolved in 4276e75